src/plugins/intel_cpu/src/nodes/generate_proposals_single_image.cpp

```diff
@@ -437,16 +439,26 @@ void GenerateProposalsSingleImageNode::execute(mkldnn::stream strm) {
             nms_cpu(pre_nms_topn, &is_dead[0], &unpacked_boxes[0], &roi_indices_[0], &num_rois, 0,
                     nms_thresh_, post_nms_topn_, coordinates_offset_);
 
-            // Only supported when batch size = 1
-            redefineOutputMemory({{num_rois, 4}, {num_rois}, {1}});
-            float *p_roi_item       = reinterpret_cast<float *>(getChildEdgesAtPort(OUTPUT_ROIS)[0]->getMemoryPtr()->GetPtr());
-            float *p_roi_score_item = reinterpret_cast<float *>(getChildEdgesAtPort(OUTPUT_SCORES)[0]->getMemoryPtr()->GetPtr());
-            uint8_t *p_roi_num_item = reinterpret_cast<uint8_t *>(getChildEdgesAtPort(OUTPUT_ROI_NUM)[0]->getMemoryPtr()->GetPtr());
-
-            auto roi_num_type = getOriginalOutputPrecisionAtPort(OUTPUT_ROI_NUM);
-            fill_output_blobs(&unpacked_boxes[0], &roi_indices_[0], p_roi_item, p_roi_score_item,
-                              p_roi_num_item, pre_nms_topn, num_rois, post_nms_topn_, roi_num_type);
+            int new_num_rois = total_num_rois + num_rois;
+            roi_item.resize(new_num_rois * 4);
+            score_item.resize(new_num_rois);
+
+            p_roi_num += roi_num_type == Precision::I32 ? n * sizeof(int32_t) : n * sizeof(int64_t);
```

Comment:
why the address of 'p_roi_num' is added before fill_output_blobs?

Reply:
fixed.